lib/container_orchestrator.rb

```diff
@@ -1,28 +1,83 @@
+require 'kubeclient'
+
 class ContainerOrchestrator
+  include_concern 'ObjectDefinition'
+
   TOKEN_FILE = "/run/secrets/kubernetes.io/serviceaccount/token".freeze
 
   def scale(deployment_config_name, replicas)
-    connection.patch_deployment_config(deployment_config_name, { :spec => { :replicas => replicas } }, ENV["MY_POD_NAMESPACE"])
+    connection.patch_deployment_config(deployment_config_name, { :spec => { :replicas => replicas } }, my_namespace)
+  end
+
+  def create_deployment_config(name)
+    definition = deployment_config_definition(name)
+    yield(definition) if block_given?
+    connection.create_deployment_config(definition)
+  rescue KubeException => e
+    raise unless e.message =~ /already exists/
```

Comment:
Should we check it if already exists rather than rescuing the error?

Reply:
Searching for a particular object that doesn't exist using the `Kubeclient` gem raises the same exception so I figured it was easier to go this way.